sycl/include/sycl/accessor.hpp

```diff
@@ -2801,6 +2804,27 @@ class __SYCL_EBO __SYCL_SPECIAL_CLASS __SYCL_TYPE(local_accessor) local_accessor
   local_accessor(const detail::AccessorImplPtr &Impl) : local_acc{Impl} {}
 #endif
 
+  // implicit conversion between const / non-const types for read only accessors
+public:
+  template <typename DataT_,
+            typename = std::enable_if_t<
+                !std::is_same_v<DataT_, DataT> && std::is_const_v<DataT> &&
+                std::is_same_v<std::remove_const_t<DataT_>,
+                               std::remove_const_t<DataT>>>>
+  local_accessor(const local_accessor<DataT_, Dimensions> &other) {
+    local_acc::impl = other.impl;
+  }
+
+  template <typename DataT_>
+  bool operator==(const local_accessor<DataT_, Dimensions> &Rhs) const {
+    return local_acc::impl == Rhs.impl;
+  }
+
+  template <typename DataT_>
+  bool operator!=(const local_accessor<DataT_, Dimensions> &Rhs) const {
+    return !(*this == Rhs);
+  }
```

Comment:
I would prefer if we move these to below the `using` aliases below.

Reply:
I'll do it shortly.